tests/unittest/_torch/multi_gpu/test_mpi_sleep_wakeup.py

```
@@ -0,0 +1,191 @@
+def _per_device_gpu_memory() -> dict:
+    """Return ``{device_index: used_bytes}`` for the current process tree
+    across all visible CUDA devices.
+
+    Querying per device (rather than summing) lets callers assert that
+    *each* participating GPU's memory changed; an aggregate sum can mask
+    a rank-1 leak if rank-0 releases enough memory to compensate.
+    """
+    try:
+        import pynvml
+        pynvml.nvmlInit()
+    except Exception:
```

Ruff (0.15.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)